llvm/lib/Transforms/IPO/AttributorAttributes.cpp

```diff
@@ -214,7 +214,7 @@ ChangeStatus clampStateAndIndicateChange<DerefState>(DerefState &S,
 static bool mayBeInCycle(const CycleInfo *CI, const Instruction *I,
                          bool HeaderOnly, Cycle **CPtr = nullptr) {
   if (!CI)
-    return true;
+    return false;
```

Comment:
I thought we need to assume the worst if certain info is not available?

Reply:
Yeah I see that now, but is there a better way to handle this to avoid simple cases from not being missed?

Reply:
Loopinfo could be use instead: https://stackoverflow.com/questions/50138271/detect-if-basicblock-is-in-a-loop ?